packages/nestjs/src/dynamic-config/dynamic-config.service.ts

```diff
@@ -0,0 +1,83 @@
+import { forwardRef, Inject, Injectable, OnModuleInit } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import NodeCache from 'node-cache';
+import { isNil } from 'lodash';
+import { isError, tryF } from 'ts-try';
+import { LoggerService } from '../logger';
+import {
+  DynamicConfigKeyType,
+  DynamicConfigOpts,
+  DynamicConfigProvider,
+} from './dynamic-config.types';
+import { DynamicConfigEntry } from './dynamic-config.entity';
+
+enum Message {
+  ErrorUpdatingDynamicConfig = 'ErrorUpdatingDynamicConfig',
+  UpdatedDynamicConfig = 'UpdatedDynamicConfig',
+}
+
+@Injectable()
+export class DynamicConfigService<Key extends DynamicConfigKeyType> implements OnModuleInit {
+  private _cache = new NodeCache();
+
+  constructor(
+    @Inject(DynamicConfigProvider.KEY_TYPE)
+    private readonly _keyType: Key,
+    @InjectRepository(DynamicConfigEntry)
+    private readonly _repo: Repository<DynamicConfigEntry>,
+    @Inject(forwardRef(() => LoggerService))
+    private readonly _logger: LoggerService<Message>,
+  ) {
+    this._logger.setContext(this.constructor.name);
+  }
+
+  async onModuleInit(): Promise<void> {
+    // keep all keys in the database to display available configuration options
+    const entries = await this._repo.find();
+    const keysToInsert = Object.values(this._keyType).filter(
+      (key) => !entries.some((e) => e.key === key),
+    );
+    await this._repo.save(keysToInsert.map((key) => new DynamicConfigEntry({ key, value: null })));
```

Comment:
what is purpose of this line ? looks like you where clearing all db values on module init ?

Reply:
what if container got restarted randomly?

Reply:
that's not what's happening here though. it inserts any non-existent values into the database